tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp

```diff
@@ -347,6 +348,42 @@ TEST(TesseractSceneGraphUnit, LoadSRDFUnit)  // NOLINT
   EXPECT_EQ(acm->getAllAllowedCollisions().size(), 0);
 }
 
+TEST(TesseractSceneGraphUnit, LoadSRDFOPWKinematicsUnit)  // NOLINT
+{
+  using namespace tesseract_scene_graph;
+  SceneGraph g;
+
+  std::string xml_string =
+      R"(<group_opw group="manipulator" a1="0.1" a2="-0.135" b="0" c1="0.615" c2="0.705" c3="0.755" c4="0.085" offsets="0.0 0.0 -1.570796 0.0 0.0 0.0" sign_corrections="1 1 1 -1 1 1"/>)";
+  tinyxml2::XMLDocument xml_doc;
+  EXPECT_TRUE(xml_doc.Parse(xml_string.c_str()) == tinyxml2::XML_SUCCESS);
+
+  tinyxml2::XMLElement* element = xml_doc.FirstChildElement("group_opw");
+  EXPECT_TRUE(element != nullptr);
+
+  GroupOPWKinematics opw_groups = parseGroupOPWKinematics(g, element, std::array<int, 3>({ 1, 0, 0 }));
+  OPWKinematicParameters opw = opw_groups["manipulator"];
+  EXPECT_NEAR(opw.a1, 0.1, 1e-8);
+  EXPECT_NEAR(opw.a2, -0.135, 1e-8);
+  EXPECT_NEAR(opw.b, 0.0, 1e-8);
+  EXPECT_NEAR(opw.c1, 0.615, 1e-8);
+  EXPECT_NEAR(opw.c2, 0.705, 1e-8);
+  EXPECT_NEAR(opw.c3, 0.755, 1e-8);
+  EXPECT_NEAR(opw.c4, 0.085, 1e-8);
+  EXPECT_NEAR(opw.offsets[0], 0.0, 1e-8);
+  EXPECT_NEAR(opw.offsets[1], 0.0, 1e-8);
+  EXPECT_NEAR(opw.offsets[2], -1.570796, 1e-8);
+  EXPECT_NEAR(opw.offsets[3], 0.0, 1e-8);
+  EXPECT_NEAR(opw.offsets[4], 0.0, 1e-8);
+  EXPECT_NEAR(opw.offsets[5], 0.0, 1e-8);
+  EXPECT_EQ(opw.sign_corrections[0], 1);
+  EXPECT_EQ(opw.sign_corrections[1], 1);
+  EXPECT_EQ(opw.sign_corrections[2], 1);
+  EXPECT_EQ(opw.sign_corrections[3], -1);
+  EXPECT_EQ(opw.sign_corrections[4], 1);
+  EXPECT_EQ(opw.sign_corrections[5], 1);
+}
+
```

Comment:
This test doesn't test when the OPW sign correction is wrong does it?

Reply:
No, it is testing the issue with the parser of the xml. I was casting to a signed char which was not working if -1. So before this change the test would not pass but now it should.

Reply:
I need to go through and create full set of unit tests for the SRDF similar to the URDF.

Reply:
If you want to merge this before doing that, that is fine with me.

Reply:
Lets wait. I should have time over the next month to go through and add these.